scripts/migrateDeprecatedLinks.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/*
+ * Quick and dirty script to update deprecated link format. That is, replace
+ * [[link label]] with [[link | label]].
+ *
+ * For more information, see https://github.com/wikkawik/WikkaWiki/issues/1207.
+ *
+ * USAGE
+ *  php scripts/migrateDeprecatedLinks.php
+ */
+class LinkMigrator {
+    public function __construct($config) {
+        $this->pages_updated = 0;
+        $this->links_migrated = 0;
+        $this->config = $config;
+        $this->db = $this->connect_to_database();
+    }
+
+    public function run() {
+        $this->backup_database($this->config);
+        $pages = $this->load_pages();
+
+        foreach ( $pages as $page ) {
+            $page['body'] = $this->replace_old_style_links($page['body']);
+            $this->update_page($page);
+        }
+
+        $this->report_results();
+    }
+
+    protected function backup_database($config) {
+        $db_archive_file = sprintf('%s-backup.%s.sql', $config['mysql_database'], date('Ymd'));
+        $mysqldump_f = 'mysqldump %s --user=%s --password=%s --single-transaction > /tmp/%s';
+
+        # Is exec the way to go here?
+        $result = exec(sprintf($mysqldump_f,
+                               $config['mysql_database'],
+                               $config['mysql_user'],
+                               $config['mysql_password'],
+                               $db_archive_file));
+
+        if ( ! $result ) {
+            printf("Database %s backed up to /tmp/%s\n", $config['mysql_database'], $db_archive_file);
+        }
```

Comment:
Just noticed that this code might be counterintuitive at a glance. `! $result` really means `0` return code which means "no error". Code could be written more clearly to indicate that. (Or just add a comment.)